relay-server/src/actors/processor.rs

```diff
@@ -1066,8 +1066,18 @@ impl EnvelopeProcessorService {
             .count();
         let mut found_profile = false;
         state.managed_envelope.retain_items(|item| match item.ty() {
-            // Drop profile without a transaction in the same envelope.
-            ItemType::Profile if transaction_count == 0 => ItemAction::DropSilently,
+            // Drop profile without a transaction in the same envelope if sampled==true.
+            // If sampled != true, then the transaction was removed due to DS (unsampled),
+            // but we still want to keep the profile for aggregations (slowest functions, etc.)
+            ItemType::Profile
+                if (transaction_count == 0)
+                    && item
+                        .get_header("sampled")
+                        .and_then(Value::as_bool)
+                        .unwrap_or(true) =>
+            {
+                ItemAction::DropSilently
+            }
```

Comment:
We **never** want to keep any profiles that came in their own envelope. This includes the old cocoa SDK which we do not want to support. So let's make sure to scope this to just profiles that came in the same envelope as the corresponding transaction.

Reply:
@Zylphrex for profiles that get sampled we still do this check.

We only do this for profiles that were non sampled due to `DS`.

@jjbayer was explaining to me (IIUC) that there is an chain of operations on the envelope as it gets processed, so if a envelope comes with transaction + profile, and then the transaction is not sampled due to DS, then the output would be the envelope without the transaction.

Gotta check with @jjbayer if there is another way to check wether the `transaction_count==0` because of DS or because it was not sent over to begin with.

Reply:
This `transaction_count` is counting the number of transaction in the envelope as it was received. So, if it's 0, it's not because of DS and we want to discard the profile no matter what.